apps/web/components/CommunityCard.tsx

```diff
@@ -1,236 +1,233 @@
-              communityAddress={communityAddress as Address}
-              registerToken={registerToken as Address}
-              connectedAccount={accountAddress as Address}
-              tokenSymbol={tokenGarden?.symbol as string}
-              registerTokenDecimals={tokenGarden?.decimals as number}
-              registerStakeAmount={registerStakeAmount as bigint}
-              memberStakedTokens={BigInt(memberStakedTokens)}
-            />
-          </div>
-        </main>
-      </div>
-    </>
-  );
-}
+"use client";
+import { useEffect, useState } from "react";
+import { RegisterMember } from "@/components";
+import {
+  UserGroupIcon,
+  BuildingOffice2Icon,
+} from "@heroicons/react/24/outline";
+import {
+  PoolCard,
+  IncreasePower,
+  CommunityProfile,
+  FormLink,
+} from "@/components";
+import { usePathname } from "next/navigation";
+import { Address, useAccount } from "wagmi";
+import {
+  TokenGarden,
+  getCommunitiesByGardenQuery,
+  isMemberDocument,
+  isMemberQuery,
+} from "#/subgraph/.graphclient";
+import { formatTokenAmount } from "@/utils/numbers";
+import { useUrqlClient } from "@/hooks/useUqrlClient";
+import { getChainIdFromPath } from "@/utils/path";
+import useSubgraphQueryByChain from "@/hooks/useSubgraphQueryByChain";
+
+export type StakesMemberType =
+  isMemberQuery["members"][number]["memberCommunity"];
+
+type CommunityQuery = NonNullable<
+  NonNullable<getCommunitiesByGardenQuery["tokenGarden"]>["communities"]
+>[number];
+
+type CommunityCardProps = CommunityQuery & {
+  tokenGarden: TokenGarden | undefined;
+} & {
+  covenantData?: { logo: string; covenant: string };
+};
+
+export function CommunityCard({
+  covenantData,
+  communityName: name,
+  id: communityAddress,
+  strategies,
+  members,
+  registerStakeAmount,
+  protocolFee,
+  communityFee,
+  tokenGarden,
+}: CommunityCardProps) {
+  const { address: accountAddress } = useAccount();
+  const [memberStakedTokens, setMemberStakedTokens] = useState<string>("0");
+  const pathname = usePathname();
+
+  const chainId = getChainIdFromPath();
+
+  const { data: result, error } = useSubgraphQueryByChain<isMemberQuery>(
+    chainId,
+    isMemberDocument,
+    {
+      me: accountAddress?.toLowerCase(),
+      comm: communityAddress.toLowerCase(),
+    },
+    {},
+    {
+      topic: "community",
+      chainId,
+      id: communityAddress,
+      type: ["add", "delete"],
+    },
+  );
+
+  useEffect(() => {
+    if (result && result.members.length > 0) {
+      const stakedTokens =
+        result.members?.[0]?.memberCommunity?.[0]?.stakedTokens;
+      const memberStakedTokens =
+        typeof stakedTokens === "string" ? stakedTokens : "0";
+
+      setMemberStakedTokens(memberStakedTokens);
+    }
+  }, [result]);
+
+  const pools = strategies ?? [];
+
+  members = members ?? [];
+  let registerToken = tokenGarden?.id ?? "0x0";
+  registerStakeAmount =
+    registerStakeAmount !== undefined ? BigInt(registerStakeAmount) : 0n;

```

Comment:
probably fixing the type based in the query also avoid this, not sure.

Reply:
Probably not my changes. Just whitespace.